run-basic-batch-operators-test.sh

```diff
@@ -64,32 +64,113 @@ cleanup() {
+	oc get csv -n openshift-operators | grep -v packageserver | grep -v NAME | awk '{print " oc delete --wait=true csv " $2 " -n openshift-operators"}' | bash || true
+	oc get csv -A | grep -v packageserver | grep -v NAME | awk '{print " oc delete --wait=true csv " $2 " -n " $1}' | bash || true
+	oc get subscriptions -A | grep -v NAME | awk '{print " oc delete --wait=true subscription " $2 " -n " $1}' | bash || true
+	oc get job,configmap -n openshift-marketplace | grep -v NAME | grep -v "configmap/kube-root-ca.crt" | grep -v "configmap/marketplace-operator-lock" | grep -v "configmap/marketplace-trusted-ca" | grep -v "configmap/openshift-service-ca.crt" | awk '{print " oc delete --wait=true " $1 " -n openshift-marketplace" }' | bash || true
 }
 
 waitDeleteNamespace() {
 	namespaceDeleting=$1
-	# Wait for the CSV to be removed
-	oc wait csv -l test-network-function.com/operator=target -n "$namespaceDeleting" --for=delete --timeout=300s || true
-
 	# Wait for the namespace to be removed
 	if [ "$namespaceDeleting" != "openshift-operators" ]; then
 
 		echo "non openshift-operators namespace = $namespaceDeleting, deleting "
-		oc wait namespace "$namespaceDeleting" --for=delete --timeout=300s || true
+		withRetry 2 10 oc wait namespace "$namespaceDeleting" --for=delete --timeout=60s || true
 		forceDeleteNamespaceIfPresent "$namespaceDeleting"
 	fi
 }
 
+# Function to execute oc command with retry
+function withRetry() {
+	local max_retries=$1
+	local timeout=$2
+	shift 2
+
+	local retries=0
+	local status=0
+	until [ "$retries" -ge "$max_retries" ]; do
+		# Execute oc command saving stdout, stderr and exit status
+		# see: https://stackoverflow.com/questions/11027679/capture-stdout-and-stderr-into-different-variables/41069638#41069638
+		unset stdout stderr status
+		eval "$(
+			(
+				#oc command
+				$"$@"
+			) \
+				2> >(
+					# shellcheck disable=SC2030
+					stderr=$(cat)
+					typeset -p stderr
+				) \
+				> >(
+					# shellcheck disable=SC2030
+					stdout=$(cat)
+					typeset -p stdout
+				)
+			# shellcheck disable=SC2030
+			status=$?
+			typeset -p status
+		)"
+		# shellcheck disable=SC2031
+		if [ "$status" == 0 ]; then
+			# If the command succeeded, break out of the loop
+			# shellcheck disable=SC2031
+			echo "$stdout"
+
+			echo "command: $*" >&2
+			echo "stderr: $stderr" >&2
+			echo "stdout: $stdout" >&2
+			echo "status: $status" >&2
+			return 0
+		fi
+		# If the command failed, increment the retry counter
+		retries=$((retries + 1))
+		# shellcheck disable=SC2031
+		echo "command: $*" >&2
+		# shellcheck disable=SC2031
+		echo "stderr: $stderr" >&2
+		# shellcheck disable=SC2031
+		echo "stdout: $stdout" >&2
+		# shellcheck disable=SC2031
+		echo "status: $status" >&2
+		echo "Retry $retries/$max_retries: Waiting for a few seconds before the next attempt..." >&2
+		sleep "$timeout"
+	done
+
+	echo "Maximum retries reached. Exiting with failure." >&2
+	return 1
+}
+
+waitClusterOk() {
+	timeoutSeconds=600
```

Comment:
Use local variables. Do not mix camel and snake cases in naming. I vote for snake case everywhere

Reply:
fixed